discord/ext/tasks/__init__.py

```diff
@@ -33,48 +48,66 @@
 
 from collections.abc import Sequence
 from discord.backoff import ExponentialBackoff
+from discord.utils import MISSING
 
 log = logging.getLogger(__name__)
 
 __all__ = (
     'loop',
 )
 
+T = TypeVar('T')
+_func = Callable[..., Awaitable[Any]]
+LF = TypeVar('LF', bound=_func)
+FT = TypeVar('FT', bound=_func)
+ET = TypeVar('ET', bound=Callable[[Any, BaseException], Awaitable[Any]])
+LT = TypeVar('LT', bound='Loop')
+
+
 class SleepHandle:
     __slots__ = ('future', 'loop', 'handle')
 
-    def __init__(self, dt, *, loop):
+    def __init__(self, dt: datetime.datetime, *, loop: asyncio.AbstractEventLoop) -> None:
         self.loop = loop
         self.future = future = loop.create_future()
         relative_delta = discord.utils.compute_timedelta(dt)
         self.handle = loop.call_later(relative_delta, future.set_result, True)
 
-    def recalculate(self, dt):
+    def recalculate(self, dt: datetime.datetime) -> None:
         self.handle.cancel()
         relative_delta = discord.utils.compute_timedelta(dt)
         self.handle = self.loop.call_later(relative_delta, self.future.set_result, True)
 
-    def wait(self):
+    def wait(self) -> asyncio.Future:
         return self.future
 
-    def done(self):
+    def done(self) -> bool:
         return self.future.done()
 
-    def cancel(self):
+    def cancel(self) -> None:
         self.handle.cancel()
         self.future.cancel()
 
 
-class Loop:
+class Loop(Generic[LF]):
```

Comment:
Is there a reason why this is generic? The underlying type seems to be a constant so it's not like it varies depending on input.